src/interface/python/instrumentation.h

```diff
@@ -0,0 +1,59 @@
+#ifndef CALI_INTERFACE_PYTHON_INSTRUMENTATION_H
+#define CALI_INTERFACE_PYTHON_INSTRUMENTATION_H
+
+#include "common.h"
+
+#include "variant.h"
+
+namespace cali {
+
+class PythonAttribute {
+public:
+  PythonAttribute(const char *name, cali_attr_type type);
+
+  PythonAttribute(const char *name, cali_attr_type type,
+                  cali_attr_properties opt);
```

Comment:
The `opt` argument should be an int. That's because you can combine multiple `cali_attr_properties` flags via `OR`. Same in all other functions that take it.

Reply:
On the C++ side, `cali_attr_properties` is an unscoped enum. As a result, per the C++ standard, it acts as an implementation-defined integral with size less than or equal to `int`. That means it's perfectly acceptable to apply an `OR` to multiple `cali_attr_properties` values.

Beyond the C++ stuff, it's actually better to use `cali_attr_properties` for the Python bindings because pybind11 does some auto-conversion stuff between Python and C++ types that could get messy if I just accepted an `int`.

One alternative that would play nice with everything is to export the properties as an `int` in pybind11 and then wrap the values in a proper Python `enum`. In that case, the Python interface could be a subclass of `enum.IntFlag`, which would provide a native way of doing `OR` logic in Python that would still convert to `int` when passed to C++.

Reply:
Ignore my comment regarding `IntFlag`. That won't really be possible because Python does not support method overloading.